app/jobs/view_illiad_loans_job.rb

```diff
@@ -12,16 +12,22 @@ def perform(webaccess_id:, type:, library_key:)
 
     illiad_loans = IlliadClient.new.send("get_loan_#{type}", webaccess_id)
 
-    html = HoldsController.render template: "#{type}/ill_#{type}", layout: false,
-                                  locals: { illiad_loans:, library_key: }
-
-    Redis.current.set("view_ill_#{type}_#{webaccess_id}", {
-      result: :success,
-      html:
-    }.to_json)
-    nil
-  rescue RuntimeError => e
-    process_failure(error_message: e.message, webaccess_id:, type:)
+    # IlliadClient usually returns an array, If we don't return an array we send the
+    # result to process_failure. We do this to avoid raising an exception in IlliadClient
+    if illiad_loans.instance_of?(Array)
```

Comment:
I take it that there are no "non-Array but still legit" results we might get?

Reply:
We should always be returning an array in the client for successful calls, even if there's one item.